File: app/src/main/java/com/dannygrove/mtls/ServerDAO.java
package com.dannygrove.mtls;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ServerDAO {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Server server);

    @Query("DELETE FROM servers where _id=:id")
    void delete(Long id);

    @Query("SELECT * FROM servers")
    LiveData<List<Server>> getServer();

    @Query("SELECT * FROM servers WHERE _id = :id LIMIT 1")
    Server getServer(Long id);

    @Update
    void update(Server server);
}

File: app/src/main/java/com/dannygrove/mtls/ServerDao.kt
package com.dannygrove.mtls

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface ServerDao {
  @Query("SELECT * FROM servers")
  fun getServers(): Flow<List<Server>>

  @Query("SELECT * FROM servers WHERE id = :id LIMIT 1")
  suspend fun getServerById(id: Int): Server?

  @Insert(onConflict = OnConflictStrategy.IGNORE)
  suspend fun addServer(server: Server)

  @Delete
  suspend fun deleteServer(server: Server)

  @Update
  fun updateServer(server: Server)
}
